### PR TITLE
6211198: ICC_Profile.getInstance(byte[]): IAE is not specified

### PR DESCRIPTION
**Pull request by mrserb**
The specification of the java.awt.color.ICC_Profile.getInstance(byte[]) is updated.
Its implementation changed over time, and different exceptions were thrown, but since JDK-8013430 always throws an IllegalArgumentException on null and invalid data.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-6211198](https://bugs.openjdk.java.net/browse/JDK-6211198): ICC_Profile.getInstance(byte[]): IAE is not specified

### Reviewers
 * [Phil Race](https://openjdk.java.net/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to 12f673c140c40c2520b6ce0336b2d265d402c1ca
 * [Pankaj Bansal](https://openjdk.java.net/census#pbansal) (@pankaj-bansal - **Reviewer**) ⚠️ Review applies to 12f673c140c40c2520b6ce0336b2d265d402c1ca

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2328/head:pull/2328`
`$ git checkout pull/2328`

### CODE REVIEW

File: src/java.desktop/share/classes/java/awt/color/ICC_Profile.java [hunk 1]
```diff
@@ -963,12 +965,7 @@ public static ICC_Profile getInstance(String fileName) throws IOException {
      *         Profile data
      */
     public static ICC_Profile getInstance(InputStream s) throws IOException {
-        byte[] profileData;
-        if ((profileData = getProfileDataFromStream(s)) == null) {
```
Comment (mrserb) [inline src/java.desktop/share/classes/java/awt/color/ICC_Profile.java:967]: This additional check was needed when the getInstance(byte[]) thrown NPE on null

**Review by prrace — APPROVED** [review]

File: src/java.desktop/share/classes/java/awt/color/ICC_Profile.java [hunk 2]
```diff
@@ -764,6 +764,8 @@ protected void finalize () {
      * @param  data the specified ICC Profile data
      * @return an {@code ICC_Profile} object corresponding to the data in the
      *         specified {@code data} array
+     * @throws IllegalArgumentException If the byte array does not contain valid
```
Comment (prrace) [inline src/java.desktop/share/classes/java/awt/color/ICC_Profile.java:767]: The spec does already say "Throws an {@code IllegalArgumentException} if the data does		     
    * not correspond to a valid ICC Profile"
So really you are just adding the @throws clause. 
You might want to consider removing the now redundant text.

**Review by pankaj-bansal — APPROVED** [review]

File: src/java.desktop/share/classes/java/awt/color/ICC_Profile.java (hunk 2 above)
Comment (mrserb) [reply src/java.desktop/share/classes/java/awt/color/ICC_Profile.java:766]: Agree. that text is removed.

**Pull request closed without merging**